src/d3d9/d3d9_device.cpp

```diff
@@ -1231,13 +1231,13 @@ namespace dxvk {
       if (unlikely(!srcIsDepth || !dstIsDepth))
         return D3DERR_INVALIDCALL;
 
-      if (unlikely(srcTextureInfo->Desc()->Discard || dstTextureInfo->Desc()->Discard))
+      if (unlikely((srcTextureInfo->Desc()->Discard || dstTextureInfo->Desc()->Discard) && StretchRectValidation))
```

Comment:
Maybe just `!m_isD3D8compatible` here instead of StretchRectValidation, and avoid the entire wrapper mess?